core/src/sql/fetch.rs

```diff
@@ -39,22 +45,66 @@ impl InfoStructure for Fetchs {
 	}
 }
 
-#[revisioned(revision = 1)]
+#[revisioned(revision = 2)]
 #[derive(Clone, Debug, Default, Eq, PartialEq, PartialOrd, Serialize, Deserialize, Hash)]
 #[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
 #[non_exhaustive]
-pub struct Fetch(pub Value);
+pub struct Fetch(
+	#[revision(end = 2, convert_fn = "convert_fetch_idiom")] pub Idiom,
+	#[revision(start = 2, default_fn = "default_value")] pub Value,
+);
+
+impl Fetch {
+	fn default_value(_revision: u16) -> Value {
+		Default::default()
+	}
+
+	fn convert_fetch_idiom(&mut self, _revision: u16, i: Idiom) -> Result<(), revision::Error> {
+		self.1 = Value::Idiom(i);
+		Ok(())
+	}
```

Comment:
@DelSkayn Could you have a look? I think that's a use case that Revision does not support yet